src/crepe/api/Vector2.h

```diff
@@ -3,29 +3,60 @@
 namespace crepe {
 
 //! 2D vector
+template <class T>
 struct Vector2 {
 	//! X component of the vector
-	double x = 0;
+	T x = 0;
 	//! Y component of the vector
-	double y = 0;
+	T y = 0;
 
 	//! Subtracts another vector from this vector and returns the result.
 	Vector2 operator-(const Vector2 & other) const;
```

Comment:
Klopt het dat je bij alle operators die een parameter meenemen alleen `Vector2` specificeert en niet `Vector2<T>`? Zoals het er nu staat zou betekenen dat je kunt rekenen met gemengde soorten vectors:
```cpp
Vector2<int> foo {1, 2};
Vector2<float> bar {3, 4};
foo -= bar; // dit mag (volgensmij?), maar is dat de bedoeling?
```

Reply:
Heel scherp!

Dit mag overigens niet, want ik heb dit wel juist genoteerd in de .hpp file:
![image](https://github.com/user-attachments/assets/28d8049a-3591-429b-af9a-5d8a0bdd2042)
Maar voor de netheid zal ik het ook even juist noteren in de .h file.